src/m365/spo/commands/list/list-view-set.ts

```diff
@@ -97,21 +103,30 @@ class SpoListViewSetCommand extends SpoCommand {
 
   #initOptionSets(): void {
     this.optionSets.push(
-      ['listId', 'listTitle'],
+      ['listId', 'listTitle', 'listUrl'],
       ['viewId', 'viewTitle']
     );
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
-    const baseRestUrl: string = `${args.options.webUrl}/_api/web/lists`;
-    const listRestUrl: string = args.options.listId ?
-      `(guid'${formatting.encodeQueryParameter(args.options.listId)}')`
-      : `/getByTitle('${formatting.encodeQueryParameter(args.options.listTitle as string)}')`;
-    const viewRestUrl: string = `/views/${(args.options.viewId ? `getById('${formatting.encodeQueryParameter(args.options.viewId)}')` : `getByTitle('${formatting.encodeQueryParameter(args.options.viewTitle as string)}')`)}`;
+    const baseRestUrl: string = `${args.options.webUrl}/_api/web/`;
+
+    let listRestUrl: string = '';
+    if (args.options.listId) {
+      listRestUrl = `lists(guid'${formatting.encodeQueryParameter(args.options.listId)}')`;
+    }
+    else if (args.options.listTitle) {
+      listRestUrl = `lists/GetByTitle('${formatting.encodeQueryParameter(args.options.listTitle as string)}')`;
+    }
+    else if (args.options.listUrl) {
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, args.options.listUrl);
+      listRestUrl = `GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')`;
+    }
+
+    const viewRestUrl: string = `/views/${(args.options.viewId ? `GetById('${formatting.encodeQueryParameter(args.options.viewId)}')` : `GetByTitle('${formatting.encodeQueryParameter(args.options.viewTitle as string)}')`)}`;
 
     try {
       const res = await spo.getRequestDigest(args.options.webUrl);
-
       const requestOptions: any = {
         url: `${baseRestUrl}${listRestUrl}${viewRestUrl}`,
```

Comment:
The `baseRestUrl` variable is not really necessary. you can just replace it with the value that's used in the declaration.